File: December 2023/Game of XOR/main.cpp
// Link: https://www.geeksforgeeks.org/problems/game-of-xor1541/1

#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    int gameOfXor(int N, int A[])
    {
        int ans = 0;
        for (int i = 0; i < N; i++)
        {
            long long res = (N - i) * (i + 1);
            if (res % 2 != 0)
            {
                ans = (ans ^ A[i]);
            }
        }
        return ans;
    }
};

int main()
{
    Solution obj;
    int A[] = {1, 2, 3};
    int N = sizeof(A) / sizeof(A[0]);
    cout << obj.gameOfXor(N, A);
    return 0;
}

File: December 2023/Game of XOR/main.py
# Link: https://www.geeksforgeeks.org/problems/game-of-xor1541/1

class Solution:
    def gameOfXor(self, N , A):
        # code here
        ans = 0
        for i, e in enumerate(A):
            if (i+1)*(N-i)%2 == 0:
                ans ^= 0
            else:
                ans ^= e
        return ans
    
if __name__ == '__main__':
    obj = Solution()
    A=[1,2,3]
    print(obj.gameOfXor(len(A), A))